app/models.py

```diff
@@ -186,3 +191,35 @@ def is_admin(self):
 @login_manager.user_loader
 def load_user(user_id):
     return User.query.get(int(user_id))
+
+
+# Agency schema
+class Agency(db.Model):
+    __tablename__ = 'agencies'
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(128), index=True)
+    is_public = db.Column(db.Boolean, default=False)
+    users = db.relationship('User', backref='agency', lazy='dynamic')
+    reports = db.relationship('IncidentReport', backref='agency',
```

Comment:
For this line and the one above it, why are you using `lazy='dynamic'`?

According to docs:

> dynamic - the attribute will return a pre-configured Query object for all read operations, onto which further filtering operations can be applied before iterating the results. See the section Dynamic Relationship Loaders for more details.

Do you really think we are going to do more filtering when we get all the users or all reports for a `Agency`? Do you think we will need filtering when we get all reports for a user or a agency? 

We may want to start with `lazy='select'` or `lazy='joined'` and then change it if we need to?